File: benchmarks/poly/jsondeserialize/jsondeserialize.py
import sys, time


def main():
    _t0 = time.perf_counter_ns()
    n = 400000
    checksum = 0
    total_name_len = 0
    for i in range(n):
        line = '{"id":' + str(i) + ',"name":"user' + str(i) + '","score":' + str(i % 100) + '}'

        id_start = line.index(":") + 1
        id_end = line.index(",")
        id_val = int(line[id_start:id_end])

        name_start = line.index('"name":"') + len('"name":"')
        name_end = line.index('"', name_start)
        name_val = line[name_start:name_end]

        score_start = line.rindex(":") + 1
        score_end = line.rindex("}")
        score_val = int(line[score_start:score_end])

        checksum = (checksum * 31 + id_val + score_val) % 1000000007
        total_name_len += len(name_val)

    print(checksum)
    print(total_name_len)
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: benchmarks/poly/jsondeserialize/jsondeserialize.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

func main() {
	t0 := time.Now()
	n := 400000
	checksum := 0
	totalNameLen := 0

	for i := 0; i < n; i++ {
		line := `{"id":` + strconv.Itoa(i) + `,"name":"user` + strconv.Itoa(i) + `","score":` + strconv.Itoa(i%100) + `}`

		idStart := strings.Index(line, ":") + 1
		idEnd := strings.Index(line, ",")
		idVal, _ := strconv.Atoi(line[idStart:idEnd])

		namePrefix := `"name":"`
		nameStart := strings.Index(line, namePrefix) + len(namePrefix)
		nameEnd := strings.Index(line[nameStart:], `"`) + nameStart
		nameVal := line[nameStart:nameEnd]

		scoreStart := strings.LastIndex(line, ":") + 1
		scoreEnd := strings.LastIndex(line, "}")
		scoreVal, _ := strconv.Atoi(line[scoreStart:scoreEnd])

		checksum = (checksum*31 + idVal + scoreVal) % 1000000007
		totalNameLen += len(nameVal)
	}

	fmt.Println(checksum)
	fmt.Println(totalNameLen)
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}
